channeldb/nodes.go

```diff
@@ -129,13 +136,13 @@ func putLinkNode(nodeMetaBucket kvdb.RwBucket, l *LinkNode) error {
 
 // DeleteLinkNode removes the link node with the given identity from the
 // database.
-func (d *DB) DeleteLinkNode(identity *btcec.PublicKey) error {
-	return kvdb.Update(d, func(tx kvdb.RwTx) error {
-		return d.deleteLinkNode(tx, identity)
+func (l *LinkNodeDB) DeleteLinkNode(identity *btcec.PublicKey) error {
+	return kvdb.Update(l.db, func(tx kvdb.RwTx) error {
+		return deleteLinkNode(tx, identity)
 	}, func() {})
 }
 
-func (d *DB) deleteLinkNode(tx kvdb.RwTx, identity *btcec.PublicKey) error {
+func deleteLinkNode(tx kvdb.RwTx, identity *btcec.PublicKey) error {
```

Comment:
Not sure what you mean. This change turns the method into a function because no reference to the `LinkNodeDB` is required here, since we have the transaction instead.